django_webapps/scripts/db_population/parsers/teacher_parsers.py

```diff
@@ -0,0 +1,159 @@
+        return TeacherSituationalStatus.NO_RESPONSE
+    if status_str in ["группы набраны"]:
+        return TeacherSituationalStatus.AWAITING_START
+    return None
+
+
+def parse_name(name_str: str) -> str:
+    return name_str
+
+
+def parse_email(email_str: str) -> str | None:
+    try:
+        validate_email(email_str)
+    except EmailNotValidError as _:
+        return None
+    return email_str
+
+
+def parse_has_experience(exp_str: str) -> bool:
+    if exp_str.lower().strip() == "да":
+        return True
+    return False
+
+
+def parse_telegram_name(tg_str: str) -> str | None:
+    tg_regex = re.compile(r"@(?=\w{5,32}\b)[a-zA-Z0-9]+(?:_[a-zA-Z0-9]+)*.*")
+    match_result = re.search(tg_regex, tg_str)
+    if match_result is not None:
+        return match_result.group(0)
+    return None
+
+
+def parse_can_give_feedback(fd_str: str) -> bool:
+    if fd_str.lower().strip() == "да":
+        return True
+    return False
+
+
+def parse_weekly_frequency(freq_str: str) -> int | None:
+    if "мин" in freq_str.lower() or "час" in freq_str.lower():
+        return 1
+    numbers = list(map(int, re.findall(r"\d+", str(freq_str))))
+    if len(numbers) > 0:
+        return max(numbers)
+    return None
+
+
+def parse_age_ranges(age_ranges_str: str) -> bool:
+    if age_ranges_str.lower().strip() == "да":
+        return True
+    return False
+
+
+def parse_language_level(level_str: str) -> list[str]:
+    language_levels = [e.value for e in LanguageLevelId]
+    if "any" in level_str.lower() or "любой" in level_str.lower():
+        return language_levels
+    regex = re.compile(rf"{'|'.join(language_levels)}")
+    result = re.findall(regex, level_str.upper())
+    return list(set(result))
+
+
+def parse_availability_slots(
+    slot_str: str, time_slots: list[tuple[int, int]] | None = None
+) -> list[tuple[int, int]]:
+    if time_slots is None:
+        time_slots = TIME_SLOTS
+
+    slot_str = slot_str.lower()
+    result = []
+    if "утро" in slot_str or "morning" in slot_str:
+        result.extend([(8, 11), (5, 8)])
+    if "вечер" in slot_str or "evening" in slot_str:
+        result.extend(
+            [
+                (11, 14),
+                (14, 17),
+            ]
+        )
+    if "день" in slot_str or "afternoon" in slot_str:
+        result.append((11, 17))
+    if len(result) > 0:
+        return result
+
+    regex = re.compile(r"(\d{1,2}):(\d{1,2})-(\d{1,2}):(\d{1,2})")
+    slot_str = slot_str.replace(" ", "").replace(".", ":")
+
+    def find_best_fits(range_to_fit: tuple[int, int]) -> list[tuple[int, int]]:
+        result = []
+        for slot in time_slots:
+            if slot[0] >= range_to_fit[0] and slot[1] <= range_to_fit[1]:
+                result.append(slot)
+        return result
+
+    re_results = re.findall(regex, slot_str)
+    if len(re_results) > 0:
+        for re_result in re_results:
+            range_to_fit = (int(re_result[0]), int(re_result[2]))
```

Comment:
is it because match groups are formed so (using parentheses) that these indices will correspond to "from_hour" and "to_hour"?
I know that this is a run-once script, but in this case I would at least ask to add a comment.